src/battle_ai_switch_items.c

```diff
@@ -606,6 +606,13 @@ static bool32 ShouldSwitchIfAbilityBenefit(u32 battler)
 
             return FALSE;
 
+        case ABILITY_ZERO_TO_HERO:
+            //Zero to Hero only works on Palafin-Zero
+            if (gBattleMons[battler].species == SPECIES_PALAFIN_ZERO
+                 && AI_DATA->mostSuitableMonId[battler] != PARTY_SIZE)
```

Comment:
Question, this implementation means Zero to Hero will only prompt switches if the AI sees it has a *good* switchin candidate. Is Zero to Hero worth switching for even if the AI *doesn't* have a good switchin candidate? If so we don't need need the mostSuitableMonId check, if it isn't worth it then everything's fine as is :)

(I know nothing about Palafin other than hearing that it needs to switch from you)

Reply:
I would say a bad switch-in is also preferable - even if the mon you send out dies because it's at low health or whatever, it still powers up Palafin. If anything, I would actually encourage some sort of viability graph where medium health is discouraged, but high HP _and_ low HP are both discouraged (either because they're a good switch-in according to the AI, or because they likely served their use and can be sacrificed to power up Palafin who is likely to be more useful in battle than them at that point)

Reply:
Yeah, I wrote this thinking that Palafin wants to switch at all costs so I'll tweak it

Reply:
Hmm if I remove this, then it will always switch and won't use a pivoting move when it should probably prioritize pivoting moves.